bigquery_etl/cli/metadata.py

```diff
@@ -0,0 +1,75 @@
+"""bigquery-etl CLI metadata command."""
+import os
+from pathlib import Path
+from typing import Optional
+
+import click
+
+from bigquery_etl.metadata.parse_metadata import DatasetMetadata, Metadata
+
+from ..cli.utils import paths_matching_name_pattern, project_id_option, sql_dir_option
+
+
+@click.group(
+    help="""
+        Commands for managing bqetl metadata.
+        \b
+
+        UNDER ACTIVE DEVELOPMENT See https://mozilla-hub.atlassian.net/browse/DENG-1381
+        """
+)
+@click.pass_context
+def metadata(ctx):
+    """Create the CLI group for the metadata command."""
+    pass
+
+
+@metadata.command(
+    help="""
+    Update table level metadata yaml files.
+   \b
+
+    Example:
+     ./bqetl metadata update ga_derived.downloads_with_attribution_v2
+    """,
+    context_settings=dict(
+        ignore_unknown_options=True,
+        allow_extra_args=True,
+    ),
+)
+@click.argument("name")
+@project_id_option()
+@sql_dir_option
+def update(name: str, sql_dir: Optional[str], project_id: Optional[str]) -> None:
+    """Update metadata yaml file."""
+    table_metadata_files = paths_matching_name_pattern(
+        name, sql_dir, project_id=project_id, files=["metadata.yaml"]
+    )
+    for table_metadata_file in table_metadata_files:
+        dataset_metadata_path = (
+            Path(table_metadata_file).parent.parent / "dataset_metadata.yaml"
+        )
+        if os.path.exists(dataset_metadata_path):
+            dataset_metadata = DatasetMetadata.from_file(dataset_metadata_path)
+            table_metadata = Metadata.from_file(table_metadata_file)
+            if table_metadata.deprecated:
+                table_metadata.workgroup_access = [
+                    dict(
+                        role="roles/bigquery.metadataViewer",
+                        members=["workgroup:deprecated"],
+                    )
+                ]
+            else:
+                if dataset_metadata.default_table_workgroup_access and (
+                    table_metadata.workgroup_access == []
+                    or table_metadata.workgroup_access is None
+                ):
+                    table_metadata.workgroup_access = (
+                        dataset_metadata.default_table_workgroup_access
+                    )
```

Comment:
This assumes that workgroups for restricted tables are set in the `metadata.yaml` or as the `default_table_workgroup_access` in the dataset metadata. Otherwise, the default will be applied which is `roles/bigquery.dataViewer`.

We could add a check based on whether `_restricted in dataset_metdata.dataset_base_acl` and not overwrite the `table_metadata.workgroup_access` for these cases. 
@whd do you have a preference here?

Reply:
I don't think we need to special case the table workgroup access metadata generation logic based on whether `dataset_base_acl` is a restricted type or not. The logic should be the same in either case, but that's assuming generation logic that looks something like:

`dataset_metadata.yaml` generation logic:
* if `dataset_metadata.yaml` doesn't exist, some default metadata are generated that include `workgroup_access`
example https://github.com/mozilla/bigquery-etl/tree/main/sql/moz-fx-data-shared-prod/burnham_derived => https://github.com/mozilla/bigquery-etl/blob/generated-sql/sql/moz-fx-data-shared-prod/burnham_derived/dataset_metadata.yaml#L7
* if `dataset_metadata.yaml` does exist, it is guaranteed to include `workgroup_access` (this might not be true but it should be, and from checking `main` is de facto true)
* if `default_table_workgroup_access` doesn't exist in `dataset_metadata.yaml`, it is set to the value of `workgroup_access` (which is guaranteed to exist) <- I haven't read the full PR but this is probably the piece that's missing

At this point all `dataset_metadata.yaml` files should have both `workgroup_access` and `default_table_workgroup_access`.

`metadata.yaml` `workgroup_access` table generation logic then looks like:
* if `workgroup_access` doesn't exist in a table's `metadata.yaml`, generate it from `default_table_workgroup_access`
* if `workgroup_access` does exist, leave the value as-is except in the the special case where a `deprecated` annotation exists that is used to override it

I do think we should have additional CI for making sure we're using restricted dataset variants correctly (https://mozilla-hub.atlassian.net/browse/DENG-1400) but for the purposes of metadata generation the logic is the same.